src/System.Net.Http/tests/FunctionalTests/ManagedHandlerTest.cs

```diff
@@ -391,5 +391,96 @@ public sealed class ManagedHandler_HttpClientHandler_ConnectionPooling_Test : Ht
                 }
             }
         }
+
+        [OuterLoop]
+        [Fact]
+        public async Task HttpConnectMethod_Succeeds()
+        {
+            using (HttpClient client = CreateHttpClient())
+            using (var listener = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp))
+            {
+                listener.Bind(new IPEndPoint(IPAddress.Loopback, 0));
+                listener.Listen(100);
+                var ep = (IPEndPoint)listener.LocalEndPoint;
+                var uri = new Uri($"http://{ep.Address}:{ep.Port}/");
+                string responseBody = "HTTP/1.1 200 Connection established\r\n\r\n";
+
+                HttpRequestMessage request = new HttpRequestMessage(new HttpMethod("CONNECT"), uri);
+                Task<HttpResponseMessage> responseTask = client.SendAsync(request);
+
+                using (Socket server = await listener.AcceptAsync())
+                using (var serverStream = new NetworkStream(server, ownsSocket: false))
+                {
+                    // Skip request headers.
+                    while (true)
```

Comment:
We should validate the target specified in the CONNECT request.  LoopbackServer helps make the parsing easier here.